File: benchmark.go
package main

import (
	"fmt"
	"math"
	"strings"
	"time"
)

func runBenchmark() {
	fmt.Println("Go Benchmark")

	// Test 1: Array operations
	startTime := time.Now()
	arr := make([]int, 0, 1000000)
	for i := 0; i < 1000000; i++ {
		arr = append(arr, i)
	}
	sum := 0
	for _, v := range arr {
		sum += v
	}
	arrayTime := float64(time.Since(startTime).Microseconds()) / 1000.0
	fmt.Printf("Array operations: %.2f ms\n", arrayTime)

	// Test 2: String operations (Optimized with strings.Builder)
	startTime = time.Now()
	var sb strings.Builder
	sb.Grow(100000) // Pre-allocate capacity
	for i := 0; i < 100000; i++ {
		sb.WriteByte('a')
	}
	// Final string is in sb but we don't need to retrieve it for the benchmark
	stringTime := float64(time.Since(startTime).Microseconds()) / 1000.0
	fmt.Printf("String operations: %.2f ms\n", stringTime)

	// Test 3: Math operations
	startTime = time.Now()
	result := 0.0
	for i := 0; i < 10000000; i++ {
		x := float64(i % 1000)
		result += math.Sqrt(x) * math.Sin(x)
		result = math.Mod(result, 1000.0)
	}
	mathTime := float64(time.Since(startTime).Microseconds()) / 1000.0
	fmt.Printf("Math operations: %.2f ms\n", mathTime)

	// Total time
	totalTime := arrayTime + stringTime + mathTime
	fmt.Printf("Total execution time: %.2f ms\n", totalTime)
}

func main() {
	runBenchmark()
}

File: benchmark.php
<?php

function runBenchmark() {
    echo "PHP Benchmark\n";
    
    // Test 1: Array operations
    $startTime = microtime(true);
    $arr = [];
    for ($i = 0; $i < 1000000; $i++) {
        $arr[] = $i;
    }
    $sumArr = array_sum($arr);
    $endTime = microtime(true);
    $arrayTime = ($endTime - $startTime) * 1000;
    echo "Array operations: " . number_format($arrayTime, 2) . " ms\n";
    
    // Test 2: String operations
    $startTime = microtime(true);
    $str = "";
    for ($i = 0; $i < 100000; $i++) {
        $str .= "a";
    }
    $endTime = microtime(true);
    $stringTime = ($endTime - $startTime) * 1000;
    echo "String operations: " . number_format($stringTime, 2) . " ms\n";
    
    // Test 3: Math operations
    $startTime = microtime(true);
    $result = 0;
    for ($i = 0; $i < 10000000; $i++) {
        $x = $i % 1000;
        $result += sqrt($x) * sin($x);
        $result = fmod($result, 1000);
    }
    $endTime = microtime(true);
    $mathTime = ($endTime - $startTime) * 1000;
    echo "Math operations: " . number_format($mathTime, 2) . " ms\n";
    
    // Total time
    $totalTime = $arrayTime + $stringTime + $mathTime;
    echo "Total execution time: " . number_format($totalTime, 2) . " ms\n";
}

// Run the benchmark
runBenchmark(); 